2017/11/29/systemd-primer-from-the-trenches.html.md

```diff
@@ -0,0 +1,514 @@
+
+Show me failed services. systemctl status will highlight at the top if units have failed, especially after a reboot.
+
+14.
+```systemctl get-default```
+
+```systemctl set-default multi-user.target``` - equivalent to runlevels 2,3 and 4
+
+to list targets run ```systemctl list-units -t target```
+
+```Runlevel```
+```Chkconfig --list```
+```telinit runlevel```
+
+Gets the run level default for the system. Not often used, if rarely, but good know when you start having boot issues or need
+to swap out to a different run level to fix things.
+
+Tip:
+* There are many tables comparing the SysV runlevels to systemd. Get your google on and search.
+* `systemctl set-default graphical.target` will set a graphical user shell. For all those that like a good desktop.
+
+15.
+```systemctl shutdown or reboot```
+
+```shutdown -r now```
+
+Reboot/shutdown the system. Personally I use ‘shutdown -r now’ still.
+
+16.
+```systemctl cat serviceName```
+
+```cd to dir, cat init file```
+
+Shows me the system service(unit) file contents and options. We will go more into these commands as we work through next week
+building and maintaining our own UNIT files.
+
+Tip:
+* systemctl cat shows all unti file information and snippets involved with unit file.
+* If you use VIM as I do, you can enable vim-systemd to help wotj syntax highlighting.
+
+17.
+```systemctl list-dependencies serviceName```
+
+What’s really depending on my process. The --all flag will show everything --before, --after, --reverse.
+
+Tip:
+* Pretty handy little dependency checker for processes.
+
+18.
+```systemctl show serviceName```
+
+Flags: -p - shows a single property of a service. 
+
+Shows more than using ‘systemctl cat servicename’. Dont forget TAB is your friend. 
+Running the ``-p`` flag and using TAB will help you.
+
+19.
+```systemctl mask serviceName```
+
+```Update-rc.d serviceName disable```
+
+Never want someone starting a service EVER!! 'Mask' is your friend and a little sneaky. See if your system admins 
+pick this one up. Good April fools day. ‘’unmask’’ to return it to its user
+
+20.
+```systemctl edit serviceName```
+
+Good flags --full
+
+```vim /etc/inittab```
+
+Yes that’s right! Edit the service file without having to cd into the directory. Now that saved me a bit of time.
+
+Tip:
+* Careful with this. The plain edit creates an override file in /etc/systemd/system to complement the original unit file. 
+* If you need to edit the original unit file use the ```--full``` flag.
+* I made mistake in my unit file or I messed up a system unti file; use `systemctl revert unit`
+
+21.
+```systemctl --output=```
+
+Good Flags verbose, --full
+
+```/etc/init.d/httpd start >httpd-output 2>&1```
+
+Particular good if you have a bad service which is playing up. Outputs a short standard message or a very verbose message
+using different flags.
+
+Tip:
+* `Journalctl -u serviceName` can help you here, but I often find it easier to reply the command and include `--output=verbose`
```

Comment:
`journalctl` should be lowercase.

Reply:
oh man the auto case really creeped back in.

Reply:
fixed